reference/src/layout/enums.md

```diff
@@ -351,90 +351,72 @@ enum Enum1<T> {
 }
 ```
 
-## Unresolved questions
-
 ### Layout of single variant enums
 
-[Issue #79.](https://github.com/rust-rfcs/unsafe-code-guidelines/issues/79)
-
-Enums that contain a **single variant** and which do not have an
-explicit `#[repr]` annotation are an important special case. Since
-there is only a single variant, the enum must be instantiated with
-that variant, which means that the enum is in fact equivalent to a
-struct. The question then is to what extent we should **guarantee**
-that the two share an equivalent layout, and also how to define the
-interaction with uninhabited types.
+**Single variant data-carrying*** enums without a `repr()` annotation have the
```

Comment:
Have we defined this precisely enough that people will not confuse this with "enums where only a single variant carris data"?

Reply:
The definition of data-carrying and fieldless enums is at the beginning of the document. This PR uses "data-carrying" and "fieldless" with the meaning of those definitions. 

Whether the definitions themselves are clear enough or not, I don't know. They were clear to me, e.g., they explicitly call out that `enum E { Varian0(Data) }` is data-carrying even if `Data` is a ZST and `E` has size zero, and they also call out that `enum F { Variant0, Variant1 }` is fieldless even though its size is not zero. 

One can always improve the definition to make them more clear, but that should probably happen in a different PR. I kind of wish that these two definitions had their place in the glossary as well.

Reply:
"data carrying" on its own isn't very clear, but when paired with "fieldless" it becomes (i think) very clear.

Reply:
I'm resolving this conversation. If someone wants to improve the definitions of "fieldless" and "data-carrying" at the beginning of this document or in the glossary PRs welcome. If someone wants to discuss renaming "data-carrying" enums to something else, please open an issue. This PR is not the place to do either.

Reply:
I think the way you are stacking adjectives here is juts not very clear. "Data-carrying enums with a single variant" would be better, IMO.